File: packages/alteration/src/main/java/fdit/alteration/core/incident/Vertex.java
package fdit.alteration.core.incident;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Vertex {

    @JsonProperty("lat")
    private Lat lat;
    @JsonProperty("lon")
    private Lon lon;

    private Long time;

    public Lat getLat() {
        return lat;
    }

    public Double getLatValue() {
        return lat.getContent();
    }

    public Boolean isLatOffset() {
        return lat.getOffset();
    }

    public void setLat(final Lat lat) {
        this.lat = lat;
    }

    public Lon getLon() {
        return lon;
    }

    public void setLon(final Lon lon) {
        this.lon = lon;
    }

    public Double getLonValue() {
        return lon.getContent();
    }

    public Boolean isLonOffset() {
        return lon.getOffset();
    }

    public Long getTime() {
        return time;
    }

    public void setTime(final Long time) {
        this.time = time;
    }
}

File: packages/fdit-ts/src/Incident/Vertex.ts
import { Lat } from './Lat'
import { Lon } from './Lon'

export class Vertex {
  private lat?: Lat
  private lon?: Lon
  private time?: number

  public getLat(): Lat | undefined {
    return this.lat
  }

  public getLatValue(): number {
    return this.lat?.getContent() ?? 0.0
  }

  public isLatOffset(): boolean {
    return this.lat?.getOffset() ?? false
  }

  public setLat(lat: Lat): void {
    this.lat = lat
  }

  public getLon(): Lon | undefined {
    return this.lon
  }

  public setLon(lon: Lon): void {
    this.lon = lon
  }

  public getLonValue(): number {
    return this.lon?.getContent() ?? 0.0
  }

  public isLonOffset(): boolean {
    return this.lon?.getOffset() ?? false
  }

  public getTime(): number | undefined {
    return this.time
  }

  public setTime(time: number): void {
    this.time = time
  }
}
